### PR TITLE
fix: [workspace]List view and icon view, last row, grid height not enough

### PR DESCRIPTION
**Pull request by liyigang1**
caused by merging code

Log: List view and icon view, last row, grid height not enough
Bug: https://pms.uniontech.com/bug-view-222951.html https://pms.uniontech.com/bug-view-223089.html https://pms.uniontech.com/bug-view-222955.html

### CODE REVIEW

File: src/plugins/filemanager/core/dfmplugin-workspace/utils/selecthelper.cpp [hunk 1]
```diff
@@ -173,12 +173,8 @@ void SelectHelper::caculateIconViewSelection(const QRect &rect, QItemSelection *
     for (int i = firstIndex; i < lastIndex; ++i) {
         const QModelIndex &index = view->model()->index(i, 0, view->rootIndex());
         // 这里就是实际绘制区域
```
Comment (Lighto-Ku) [inline src/plugins/filemanager/core/dfmplugin-workspace/utils/selecthelper.cpp]: 没用的注释可以删了

File: src/plugins/filemanager/core/dfmplugin-workspace/views/fileview.cpp [hunk 2]
```diff
@@ -687,6 +688,23 @@ void FileView::updateViewportContentsMargins(const QSize &itemSize)
     viewport()->setContentsMargins(margin, 0, margin, 0);
 }
 
+bool FileView::selected(const QRect &actualRect, const QModelIndex &index)
```
Comment (Lighto-Ku) [inline src/plugins/filemanager/core/dfmplugin-workspace/views/fileview.cpp]: 这个函数命名建议改下，从内容上看函数的功能是计算item内的rect是否被框在选择框内，取个直观一点的函数名更好理解些。

File: src/plugins/filemanager/core/dfmplugin-workspace/utils/selecthelper.cpp (hunk 1 above)
Comment (liyigang1) [reply src/plugins/filemanager/core/dfmplugin-workspace/utils/selecthelper.cpp]: 好的

File: src/plugins/filemanager/core/dfmplugin-workspace/views/fileview.cpp (hunk 2 above)
Comment (liyigang1) [reply src/plugins/filemanager/core/dfmplugin-workspace/views/fileview.cpp]: 好的

**Review by Lighto-Ku — APPROVED** [review]

**Review by max-lvs — APPROVED** [review]

**Pull request merged**